### PR TITLE
G0 7507: datepicker in ledger and dashboard customer vendor module

### PR DESCRIPTION
**Pull request by rohit-walkover**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/app/ledger/components/advance-search/advance-search.component.ts [hunk 1]
```diff
@@ -400,4 +435,50 @@ export class AdvanceSearchModelComponent implements OnInit, OnDestroy, OnChanges
...
+        if (value && value.name) {
+            this.selectedRangeLabel = value.name;
+        }
+        this.hideGiddhDatepicker();
+        if (value && value.startDate && value.endDate) {
+            this.selectedDateRange = { startDate: moment(value.startDate), endDate: moment(value.endDate) };
+            this.selectedDateRangeUi = moment(value.startDate).format(GIDDH_NEW_DATE_FORMAT_UI) + " - " + moment(value.endDate).format(GIDDH_NEW_DATE_FORMAT_UI);
+            this.fromDate = moment(value.startDate).format(GIDDH_DATE_FORMAT);
+            this.toDate = moment(value.endDate).format(GIDDH_DATE_FORMAT);
+            let bsDaterangepicker = this.advanceSearchForm.get('bsRangeValue');
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/ledger/components/advance-search/advance-search.component.ts:481]: this is unused and can be removed

Comment (rohit-walkover) [reply apps/web-giddh/src/app/ledger/components/advance-search/advance-search.component.ts]: Done

File: apps/web-giddh/src/app/home/components/cr-dr-list/cr-dr-list.component.ts [hunk 2]
```diff
@@ -16,6 +18,20 @@ import { GIDDH_DATE_FORMAT } from '../../../shared/helpers/defaultDateFormat';
 })
 
 export class CrDrComponent implements OnInit, OnDestroy {
+
+    @ViewChild('datepickerTemplate') public datepickerTemplate: ElementRef;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/home/components/cr-dr-list/cr-dr-list.component.ts:22]: Missing JS doc

File: apps/web-giddh/src/app/home/components/cr-dr-list/cr-dr-list.component.ts [hunk 3]
```diff
@@ -140,4 +169,50 @@ export class CrDrComponent implements OnInit, OnDestroy {
...
+    public showGiddhDatepicker(element: any): void {
+        if (element) {
+            this.dateFieldPosition = this.generalService.getPosition(element.target);
+        }
+        this.modalRef = this.modalService.show(
+            this.datepickerTemplate,
+            Object.assign({}, { class: 'modal-lg giddh-datepicker-modal', backdrop: false, ignoreBackdropClick: this.isMobileScreen })
+        );
+    }
+    /**v
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/home/components/cr-dr-list/cr-dr-list.component.ts:187]: Remove character **v** in JS doc

File: apps/web-giddh/src/app/home/components/cr-dr-list/cr-dr-list.component.ts [hunk 4]
```diff
@@ -140,4 +169,50 @@ export class CrDrComponent implements OnInit, OnDestroy {
             this.getAccountsReport();
         }
     }
+
+    /**
+    * This will show the datepicker
+    *
+    * @memberof ProfitLossComponent
+    */
+    public showGiddhDatepicker(element: any): void {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/home/components/cr-dr-list/cr-dr-list.component.ts:178]: Update JS doc

File: apps/web-giddh/src/app/home/components/cr-dr-list/cr-dr-list.component.ts [hunk 5]
```diff
@@ -140,4 +169,50 @@ export class CrDrComponent implements OnInit, OnDestroy {
...
+    /**v
+    * This will hide the datepicker
+    *
+    * @memberof ProfitLossComponent
+    */
+    public hideGiddhDatepicker(): void {
+        this.modalRef.hide();
+    }
+
+    /**
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/home/components/cr-dr-list/cr-dr-list.component.ts:196]: Format JS doc

File: apps/web-giddh/src/app/home/components/profit-loss/profile-loss.component.ts [hunk 6]
```diff
@@ -20,14 +20,36 @@ import {
...
 @Component({
     selector: 'profit-loss',
     templateUrl: 'profit-loss.component.html',
-    styleUrls: ['../../home.component.scss' , './profit-loss.component.scss'],
+    styleUrls: ['../../home.component.scss', './profit-loss.component.scss'],
 })
 
 export class ProfitLossComponent implements OnInit, OnDestroy {
+
+    @ViewChild('datepickerTemplate') public datepickerTemplate: ElementRef;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/home/components/profit-loss/profile-loss.component.ts:34]: Missing JS doc

File: apps/web-giddh/src/app/home/components/profit-loss/profile-loss.component.ts [hunk 7]
```diff
@@ -220,9 +252,64 @@ export class ProfitLossComponent implements OnInit, OnDestroy {
...
     public refreshChart() {
         this.requestInFlight = true;
         this.plRequest.refresh = true;
         this.store.dispatch(this.tlPlActions.GetProfitLoss(_.cloneDeep(this.plRequest)));
     }
+
+    /**
+    * This will show the datepicker
+    *
+    * @memberof ProfitLossComponent
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/home/components/profit-loss/profile-loss.component.ts:269]: Update JS doc @params is missing

File: apps/web-giddh/src/app/home/components/profit-loss/profile-loss.component.ts [hunk 8]
```diff
@@ -220,9 +252,64 @@ export class ProfitLossComponent implements OnInit, OnDestroy {
...
+     * This will hide the datepicker
+     *
+     * @memberof ProfitLossComponent
+     */
+    public hideGiddhDatepicker(): void {
+        this.modalRef.hide();
+    }
+
+    /**
+       * Call back function for date/range selection in datepicker
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/home/components/profit-loss/profile-loss.component.ts:291]: Format JS DOC

File: apps/web-giddh/src/app/ledger/components/advance-search/advance-search.component.ts [hunk 9]
```diff
@@ -400,4 +437,54 @@ export class AdvanceSearchModelComponent implements OnInit, OnDestroy, OnChanges
         this.destroyed$.next(true);
         this.destroyed$.complete();
     }
+
+    /**
+   * This will show the datepicker
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/ledger/components/advance-search/advance-search.component.ts:442]: Format JS doc

File: apps/web-giddh/src/app/ledger/components/advance-search/advance-search.component.scss [hunk 10]
```diff
@@ -46,15 +46,30 @@
...
     .ledger-head thead tr th.active {
         display: block !important;
         width: 100%;
     }
-    .ledger_book .h-table tbody .ledger-row td {
+    .ledger-book .h-table tbody .ledger-row td {
         position: relative;
         padding-left: 80px !important;
     }
+    .Advance-Search{
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/ledger/components/advance-search/advance-search.component.scss:60]: Do use correct naming convention for classes

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.scss [hunk 11]
```diff
@@ -7,7 +7,7 @@
     height: 30px;
 }
 
-.ledgHead.entry-popup table td.select2-ledger>sh-select {
+.ledg-head.entry-popup table td.select2-ledger>sh-select {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.scss:10]: Consider naming **ledger-head**

File: apps/web-giddh/src/app/ledger/ledger.component.html [hunk 12]
```diff
@@ -1,81 +1,108 @@
...
                 <!--type="text" ng-model="cDate" options="opts" required="">-->
             </div>
             <!-- endregion -->
-
-            <div class="col-sm-4 col-xs-12 text-right C-flex-end search-options">
+            <div class="col-md-4 col-sm-4 col-xs-12 text-right C-flex-end search-options" *ngIf="!isMobileScreen">
                 <div class="d-flex xs-width-100 ledger-head-top">
 
-                    <div *ngIf="isLedgerAccountAllowsMultiCurrency" class="advancSearchBlock">
+                    <div *ngIf="isLedgerAccountAllowsMultiCurrency" class="advanc-search-block">
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/ledger/ledger.component.html:100]: Do use complete names in classes **advance-search-block**

File: apps/web-giddh/src/app/ledger/ledger.component.ts [hunk 13]
```diff
@@ -129,6 +93,7 @@ export class LedgerComponent implements OnInit, OnDestroy {
     @ViewChild('bulkActionConfirmationModal') public bulkActionConfirmationModal: ModalDirective;
     @ViewChild('bulkActionGenerateVoucherModal') public bulkActionGenerateVoucherModal: ModalDirective;
     @ViewChild('ledgerSearchTerms') public ledgerSearchTerms: ElementRef;
+    @ViewChild('datepickerTemplate') public datepickerTemplate: ElementRef;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/ledger/ledger.component.ts:96]: Missing JS doc

File: apps/web-giddh/src/app/ledger/ledger.component.ts [hunk 14]
```diff
@@ -144,6 +109,7 @@ export class LedgerComponent implements OnInit, OnDestroy {
     public uploadInput: EventEmitter<UploadInput>;
     public fileUploadOptions: UploaderOptions;
     public isFileUploading: boolean = false;
+    public isMobileScreen: boolean = true;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/ledger/ledger.component.ts:112]: Missing JS doc

File: apps/web-giddh/src/app/ledger/ledger.component.ts [hunk 15]
```diff
@@ -155,6 +121,7 @@ export class LedgerComponent implements OnInit, OnDestroy {
     public isCompanyCreated$: Observable<boolean>;
     public debitSelectAll: boolean = false;
     public creditSelectAll: boolean = false;
+    public debitCreditSelectAll: boolean = false;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/ledger/ledger.component.ts:124]: Missing JS doc

File: apps/web-giddh/src/app/ledger/ledger.component.ts [hunk 16]
```diff
@@ -202,6 +169,29 @@ export class LedgerComponent implements OnInit, OnDestroy {
     public columnarReportExportRequest: ExportLedgerRequest;
     private destroyed$: ReplaySubject<boolean> = new ReplaySubject(1);
     private accountUniquename: any;
+    public allTransactionsList: any[] = [];
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/ledger/ledger.component.ts:172]: Missing JS doc

File: apps/web-giddh/src/app/ledger/ledger.component.ts [hunk 17]
```diff
@@ -277,22 +264,28 @@ export class LedgerComponent implements OnInit, OnDestroy {
...
 
     public selectedDate(value: any) {
-        this.needToShowLoader = false;
-        let from = moment(value.picker.startDate, 'DD-MM-YYYY').toDate();
-        let to = moment(value.picker.endDate, 'DD-MM-YYYY').toDate();
 
-        // if ((this.advanceSearchRequest.dataToSend.bsRangeValue[0] !== from) || (this.advanceSearchRequest.dataToSend.bsRangeValue[1] !== to)) {
+        this.selectedRangeLabel = "";
+
+        if(value && value.name) {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/ledger/ledger.component.ts:270]: Run formatter **only on this function**

File: apps/web-giddh/src/app/ledger/components/advance-search/advance-search.component.scss (hunk 10 above)
Comment (rohit-walkover) [reply apps/web-giddh/src/app/ledger/components/advance-search/advance-search.component.scss:60]: @shalinee-walkover Please change it everywhere where -ever it is used

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.scss (hunk 11 above)
Comment (rohit-walkover) [reply apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.scss:10]: @shalinee-walkover  please check it

**Pull request merged**